crates/wasm-smith/src/lib.rs

```diff
@@ -2458,6 +2459,21 @@ struct Outer {
     modules: Vec<Rc<ModuleType>>,
 }
 
+/// A container for the kinds of instructions that wasm-smith is allowed to
+/// emit. Usage:
+/// ```
+/// # use wasm_smith::{InstructionKinds, InstructionKind};
+/// InstructionKinds::new(InstructionKind::Numeric | InstructionKind::Memory);
+/// ```
+#[derive(Clone, Copy, Debug, Default, Deserialize)]
+pub struct InstructionKinds(pub(crate) FlagSet<InstructionKind>);
+impl InstructionKinds {
+    /// Create a new container.
+    pub fn new(kinds: impl Into<FlagSet<InstructionKind>>) -> Self {
```

Comment:
Technically `FlagSet` is still in the public API given this bound on the argument

Reply:
Yeah, see above.